server/sonar-server/src/main/java/org/sonar/server/usergroups/ws/UpdateAction.java

```diff
@@ -65,61 +74,80 @@ public void define(NewController context) {
+      .setDescription(String.format("New optional description for the group. A group description cannot be larger than %d characters. " +
+        "If value is not defined, then description is not changed.", DESCRIPTION_MAX_LENGTH))
       .setExampleValue("Default group for new users");
   }
 
   @Override
   public void handle(Request request, Response response) throws Exception {
     userSession.checkLoggedIn().checkPermission(GlobalPermissions.SYSTEM_ADMIN);
 
-    Long groupId = request.mandatoryParamAsLong(PARAM_ID);
-    String name = request.param(PARAM_NAME);
-    String description = request.param(PARAM_DESCRIPTION);
-
-    DbSession dbSession = dbClient.openSession(false);
-    try {
+    try (DbSession dbSession = dbClient.openSession(false)) {
+      long groupId = request.mandatoryParamAsLong(PARAM_GROUP_ID);
       GroupDto group = dbClient.groupDao().selectById(dbSession, groupId);
-      if (group == null) {
-        throw new NotFoundException(String.format("Could not find a user group with id '%s'.", groupId));
-      }
-      String oldName = group.getName();
-      if (name != null) {
-        groupUpdater.checkNameIsUnique(name, dbSession);
-        groupUpdater.validateName(name);
-        group.setName(name);
-        updateDefaultGroupIfNeeded(dbSession, oldName, name);
+      checkFound(group, "Could not find a user group with id '%s'.", groupId);
+      Optional<OrganizationDto> org = dbClient.organizationDao().selectByUuid(dbSession, group.getOrganizationUuid());
+      checkFoundWithOptional(org, "Could not find organization with id '%s'.", group.getOrganizationUuid());
+
+      boolean changed = false;
+      String newName = request.param(PARAM_GROUP_NAME);
+      if (newName != null) {
+        changed = true;
+        UserGroupValidation.validateGroupName(newName);
+        support.checkNameDoesNotExist(dbSession, group.getOrganizationUuid(), newName);
+
+        String oldName = group.getName();
+        group.setName(newName);
+        updateDefaultGroupIfNeeded(dbSession, org.get(), oldName, newName);
       }
+
+      String description = request.param(PARAM_GROUP_DESCRIPTION);
       if (description != null) {
-        groupUpdater.validateDescription(description);
-        group.setDescription(description);
+        changed = true;
+        group.setDescription(support.validateDescription(description));
+      }
+
+      if (changed) {
+        dbClient.groupDao().update(dbSession, group);
+        dbSession.commit();
       }
-      dbClient.groupDao().update(dbSession, group);
-      dbSession.commit();
-
-      JsonWriter json = response.newJsonWriter().beginObject();
-      groupUpdater.writeGroup(json, group, dbClient.groupMembershipDao().countUsersByGroups(dbSession, singletonList(groupId)).get(group.getName()));
-      json.endObject().close();
-    } finally {
-      closeQuietly(dbSession);
+
+      writeResponse(dbSession, request, response, org.get(), group);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Call "org.isPresent()" before accessing the value. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS3655)